src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -279,7 +279,8 @@ public override Task SendAsync(ArraySegment<byte> buffer, WebSocketMessageType m
 
             try
             {
-               WebSocketValidate.ThrowIfInvalidState(_state, _disposed, s_validSendStates);
+                ThrowIfDisposed();
```

Comment:
I don't understand: the caller wouldn't have direct access to the ManagedWebSocket, which would be wrapped by another WebSocket-derived your that would do the checks and then delegate. Can you elaborate on how the situation your describe could then happen?

That said, it's not a big deal in this particular case.